spyder/plugins/outlineexplorer/widgets.py

```diff
@@ -905,80 +954,63 @@ def stop_symbol_services(self, language):
-                                           _("Show/hide outline explorer"),
-                                           icon='outline_explorer_vis',
-                                           toggled=self.toggle_visibility)
-        self.visibility_action.setChecked(True)
-
-        btn_layout = QHBoxLayout()
-        for btn in self.setup_buttons():
-            btn.setAutoRaise(True)
-            btn.setIconSize(QSize(16, 16))
-            btn_layout.addWidget(btn)
-        if options_button:
-            btn_layout.addStretch()
-            btn_layout.addWidget(self.loading_widget, Qt.AlignRight)
-            btn_layout.addWidget(options_button, Qt.AlignRight)
-
-        layout = create_plugin_layout(btn_layout, self.treewidget)
+        self.treewidget.header().hide()
+
+        layout = QHBoxLayout()
+        layout.addWidget(self.treewidget)
         self.setLayout(layout)
 
-    @Slot(bool)
-    def toggle_visibility(self, state):
-        self.setVisible(state)
-        current_editor = self.treewidget.current_editor
-        if current_editor is not None:
-            current_editor.give_focus()
-            if state:
-                self.is_visible.emit()
-
-    def setup_buttons(self):
-        """Setup the buttons of the outline explorer widget toolbar."""
-        self.fromcursor_btn = create_toolbutton(
-            self, icon=ima.icon('fromcursor'), tip=_('Go to cursor position'),
+    # ---- PluginMainWidget API
+    # ------------------------------------------------------------------------
+    def get_focus_widget(self):
+        """Define the widget to focus."""
+        return self.treewidget
+
+    def get_title(self):
+        """Return the title of the plugin tab."""
+        return _("Outline")
+
+    def setup(self):
+        """Performs the setup of plugin's menu and actions."""
+        # Toolbar buttons
+        toolbar = self.get_main_toolbar()
+        fromcursor_btn = self.create_toolbutton(
+            OutlineExplorerToolbuttons.GoToCursor, icon=ima.icon('fromcursor'),
+            tip=_('Go to cursor position'),
             triggered=self.treewidget.go_to_cursor_position)
 
-        buttons = [self.fromcursor_btn]
-        for action in [self.treewidget.collapse_all_action,
-                       self.treewidget.expand_all_action,
-                       self.treewidget.restore_action,
-                       self.treewidget.collapse_selection_action,
-                       self.treewidget.expand_selection_action]:
-            buttons.append(create_toolbutton(self))
-            buttons[-1].setDefaultAction(action)
-        return buttons
+        for item in [fromcursor_btn,
+                     self.treewidget.collapse_all_action,
+                     self.treewidget.expand_all_action,
+                     self.treewidget.restore_action,
+                     self.treewidget.collapse_selection_action,
+                     self.treewidget.expand_selection_action]:

```

Comment:
I think we need to accommodate half of this buttons in the main toolbar and half in a secondary toolbar- That's because the changes in PR #14933 are going this pane too wide.

What do you think?